src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/RsaSecurityKey.cs

```diff
@@ -0,0 +1,281 @@
+﻿using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Xml;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    sealed public class RsaSecurityKey : AsymmetricSecurityKey
+    {
+        PrivateKeyStatus privateKeyStatus = PrivateKeyStatus.AvailabilityNotDetermined;
+        readonly RSA rsa;
+
+        public RsaSecurityKey(RSA rsa)
+        {
+            if (rsa == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("rsa");
+
+            this.rsa = rsa;
+        }
+
+        public override int KeySize
+        {
+            get { return this.rsa.KeySize; }
+        }
+
+        public override byte[] DecryptKey(string algorithm, byte[] keyData)
+        {
+            switch (algorithm)
+            {
+                case SecurityAlgorithms.RsaV15KeyWrap:
+                    return EncryptedXml.DecryptKey(keyData, rsa, false);
+                case SecurityAlgorithms.RsaOaepKeyWrap:
+                    return EncryptedXml.DecryptKey(keyData, rsa, true);
+                default:
+                    if (IsSupportedAlgorithm(algorithm))
+                        return EncryptedXml.DecryptKey(keyData, rsa, false);
+
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new CryptographicException(SR.Format("UnsupportedAlgorithmForCryptoOperation",
```

Comment:
I've provided the resource strings for identity model now so you can fix up all the error message in this file.

Reply:
moved some to resource and using.. thanks for sharing..